crates/subspace-service/src/lib.rs

```diff
@@ -318,6 +321,22 @@ where
                         return false;
                     }
 
+                    // Check for fast-sync state
+                    {
+                        if let Some(state_block_number) = fast_sync_state.lock().as_ref() {
+                            let parent_block_number = *parent_header.number();
+                            if parent_block_number < *state_block_number {
+                                debug!(
+                                    %parent_block_number,
+                                    %state_block_number,
+                                    "Skipped PoT verification because of the fast sync state block."
+                                );
+
+                                return true;
+                            }
+                        }
+                    }
```

Comment:
Why are we skipping PoT verification here? PR description "PoT check update to support fast-sync" is not sufficient to understand why this is required and I do not remember discussing this.

Reply:
I can guess from the code that this is some kind of special case, but I don't understand what that special case is, what issues it causes and why you decided to fix it this way out of a set of possibilities. I sometimes can infer those things, but you'll have to help me understand it in this case.

Reply:
But we always import blocks with state, I'm not sure I'm following what you're saying.

Reply:
So the reason you had to add it here is the side effect of incorrectly importing two raw blocks instead of one (the first one being without state). This is also the reason a more advanced version of https://github.com/subspace/subspace/pull/2767 failed: there existed block without state that should not have been there. The fact that that extra block is present will also cause archiver to fail to restart if interrupted right after insertion of the first block into database.

Basically insertion of two raw blocks is incorrect and should be fixed.

Once fixed, the failure will happen at the beginning of the function when attempting to read parent header and should be handled there accordingly.

I checked the code and I believe we should never ever have invalid votes included in the block and parent hash always comes from runtime's state and not user input, thus it should be safe to return `true` from this function in that special case.

Reply:
Fixed in https://github.com/subspace/subspace/pull/2779